### ISSUE TITLE
Any plans to move away from method channels?

### ISSUE THREAD

**Issue by ItsVeridian** [jakky1/video_player_win#52]
I think that because this plugin is still using method channels, it's creating stutter on the UI thread when videos load and dispose despite being async, probably because of the thread merge:

https://www.youtube.com/watch?v=miW7vCmQwnw

We probably need to make a native thread(s) or isolate(s) to fix the issue as well.

**Comment by jakky1** [comment]
I've tried fixing this issue. 
Please test the new version on GitHub and let me know if it resolves your problem.

Remember to edit your `pubspec.yaml` for testing:
```yaml
dependencies:
  video_player_win:
    git:
      url: https://github.com/jakky1/video_player_win.git
      ref: fix_open_dispose_ui_jank
```

**Comment by ItsVeridian** [comment]
It still seems to cause some jank. I'm trying to fix it but i can't find a simple way to make openVideo static for an ffi call.

**Comment by ItsVeridian** [comment]
Actually I just double-checked the code. It seems your thread is actually detached. Not sure how it works with the method channel but it's strange there's still jank after that change.

**Comment by jakky1** [comment]
Since Flutter's Texture currently can only be accessed via `MethodChannel`, pure `FFI` won't cut it.

Also, `thread.detach()` doesn't block the UI thread at all. 
If you're curious, try timing the first-to-last row computation inside `HandleMethodCall()`—I've tested it on the latest versions, and it's consistently 0ms.

**Comment by jakky1** [comment]
<img width="600" height="600" alt="Image" src="https://github.com/user-attachments/assets/83372fea-738e-423c-b0d1-3f913ae6b366" />

=================

After some surveys and experiments, I've concluded this issue is currently unsolvable. 

I also tested other similar video player packages, and they exhibit the exact same problem.

Running in `profiling` mode, you can spot UI jank on the `DevTools Performance page` during video reloads. On my PC, that frame spends the vast majority (29.6ms) on `"Shader compilation"` (see image above).

My hunch: Flutter recompiles shaders every time a texture is created. Per the official docs (https://liudonghua123.github.io/flutter_website/perf/shader/), this blocks the Flutter `raster thread` in `Skia`, causing the jank. Switching to `Impeller` is the fix—but Windows is `Skia`-only for now. 

I just tried forcing `Impeller` on, and it broke the app entirely. We'll have to wait for official Windows `Impeller` support.

Interestingly, I noticed that after reloading the video 3+ times, subsequent reloads skip the shader compilation jank entirely.

If you have no further questions, I'll merge yesterday's changes into the new version and close this issue.

**Comment by jakky1** [comment]
I just tested the unmodified `video_player_win` v3.2.2 in `profile mode`, monitoring the `DevTools Performance page`. No noticeable difference in UI jank before/after yesterday's patch.

In other words, that patch did nothing for the UI jank. Nearly all of UI jank stem from shader compilation.

Quick correction: I found UI jank from `shader compilation` only hits on the `first video load`, and the `first pause/reload` afterward. Subsequent reloads are smooth—no jank. Can you repro this in your setup?

Since the patch did nothing for the UI jank, I'm dropping it and won't release a new version.
